src/validation.cpp

```diff
@@ -2284,14 +2281,53 @@ bool CChainState::FlushStateToDisk(
         CoinsCacheSizeState cache_state = GetCoinsCacheSizeState(&m_mempool);
         LOCK(cs_LastBlockFile);
         if (fPruneMode && (fCheckForPruning || nManualPruneHeight > 0) && !fReindex) {
+            // Previously, we called the global function ::ChainActive() in
+            // FindFilesToPrune{,Manual} to get the tip height and to determine
+            // whether or not a tip even exists. Now, we are simply passing in
+            // m_chain.Height() (which returns -1 if the tip doesn't exist). To
+            // make sure we're not changing behaviour, let's check that
+            // ::ChainActive() is the same object as m_chain (not just
+            // identical).
+            //
+            // This comment and the following assert will be removed in a
+            // subsequent commit, as they're just meant to demonstrate
+            // correctness (you can run tests against it and see that nothing
+            // exit unexpectedly).
+            assert(std::addressof(::ChainActive()) == std::addressof(m_chain));
```

Comment:
In commit "validation: Move FindFilesToPrune{,Manual} to BlockManager" (84d983656f1927955e516a94366b7c216bacded3)

Nice way of verifying, and didn't know about std::addressof!